pantheon/src/main/java/tech/pegasys/pantheon/cli/operator/OperatorSubCommand.java

```diff
@@ -0,0 +1,319 @@
+        createFileAndWrite(nodeDirectoryPath, privateKeyFileName, privateKey.toString());
+      }
+    }
+
+    /**
+     * Computes RLP encoded exta data from pre filled list of addresses.
+     *
+     * @throws IOException
+     */
+    private void processExtraData() throws IOException {
+      LOG.info("Generating genesis extra data.");
+      final String jsonAddresses = Json.encodePrettily(addressesForGenesisExtraData);
+      final String extraData = IbftExtraData.fromJsonAddresses(jsonAddresses).encode().toString();
+      genesisConfig.put("extraData", extraData);
+    }
+
+    private void createFileAndWrite(
+        final Path directory, final String fileName, final String content) throws IOException {
+      final Path filePath = directory.resolve(fileName);
+      Files.createFile(filePath);
+      Files.write(filePath, content.getBytes(UTF_8));
+    }
+
+    /**
+     * Parses the root configuration file and related sub elements.
+     *
+     * @throws IOException
+     */
+    private void parseConfig() throws IOException {
+      final String configString =
+          Resources.toString(configurationFile.toPath().toUri().toURL(), UTF_8);
+      operatorConfig = new JsonObject(configString);
+      genesisConfig = operatorConfig.getJsonObject("genesis");
+      blockchainConfig = operatorConfig.getJsonObject("blockchain");
+      nodesConfig = blockchainConfig.getJsonObject("nodes");
+      generateNodesKeys = nodesConfig.getBoolean("generate", false);
+    }
+
+    /**
+     * Creates the output directory if not exists.
+     *
+     * @throws IOException
+     */
+    private void createOutputDirectory() throws IOException {
+      final Path outputDirectoryPath = outputDirectory.toPath();
+
+      if (outputDirectory.exists() && outputDirectory.isDirectory()) {
+        try (Stream<Path> stream = Files.walk(outputDirectoryPath)) {
+          stream.sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
+          deleteIfExists(outputDirectoryPath);
+        }
+      }
+      Files.createDirectory(outputDirectoryPath);
+      keysDirectory = outputDirectoryPath.resolve("keys");
+      Files.createDirectory(keysDirectory);
+    }
+
+    /**
+     * Write the content of the genesis to the output file.
+     *
+     * @param directory The directory to write the file to.
+     * @param fileName The name of the output file.
+     * @param genesis The genesis content.
+     * @throws IOException
+     */
+    private void writeGenesisFile(
+        final File directory, final String fileName, final JsonObject genesis) throws IOException {
+      final Path genesisPath = directory.toPath().resolve(fileName);
+      deleteIfExists(genesisPath);
+      LOG.info("Writing genesis file.");
+      Files.write(genesisPath, genesis.encodePrettily().getBytes(UTF_8));
+    }
+  }
+
+  private static boolean isAnyDuplicate(String... values) {
+    final Map<String, Integer> occurences = new HashMap<>();
```

Comment:
(Optional) This could be simplified a bit by just using a HashSet and checking for membership.

Reply:
Done.